apps/server/src/app/export/msword.ts

```diff
@@ -40,6 +55,59 @@ export class MSWordExporter {
         text: implementationGuide.title || implementationGuide.name,
         heading: HeadingLevel.HEADING_1
       }));
+
+
+      this.body.push(new Paragraph({
+        text: "Description:" + implementationGuide.description
+      }));
+      this.body.push(new Paragraph({
+        text: "Publisher:" + implementationGuide.publisher
+      }));
+      this.body.push(new Paragraph({
+        text: "Copyright:" + implementationGuide.copyright
+      }));
+
+      this.body.push(new Paragraph({
+        text: ""
+      }));
+
+      bundle.entry
+        .filter(entry => entry.resource && (entry.resource.resourceType === 'StructureDefinition' ||
```

Comment:
I suggest creating separate filters for each resourceType. Example:
```
this.body.push(new Paragraph({ text: 'Profiles' });
bundle.entry.filter(entry => entry.resource && entry.resourceType === 'StructureDefinition')
  .forEach(entry => {
    const structureDefinition: StructureDefinition = <StructureDefinition> entry.resource;   // we can do this because we know entry.resource will be a StructureDefinition because of the filter
    this.body.push(new Paragraph({ text: `Name: ${structureDefinition.name` });    // name is required, so we can expect it to be populated with a value
    if (structureDefinition.title) {
      this.body.push(new Paragraph({ text: `Title: ${structureDefinition.title` });
    }

    if (structureDefinition.differential) {
      (structureDefinition.differential.element || []).forEach((element, eIndex) => {
        this.body.push(new Paragraph({ text: `Element ${eIndex + 1}` });
        this.body.push(new Paragraph({ text: `Path: ${element.path}` });
        // more fields...
      });
  });

  // filter and forEach on "ValueSet". output properties specific to value set

  // filter and forEach on "CodeSystem". output properties specific to code system

  // filter and forEach on "CapabilityStatement". output properties specific to capability statement

  // etc.
```

Reply:
@seanmcilvenna this is where I had issue with undefined properties name and title of StructureDefinition.  Remember, I asked this question couple of days ago. Bottomline is receiving compilation error that name and title are not properties of StructureDefinition.

Reply:
with the way the code is now, I can see the errors about name and title... you aren't casting `entry.resource` to anything, so the compiler just thinks it's a DomainResource, which doesn't have any resource-specific properties on it, such as name and title. You have to determine what type of resource it is by checking `entry.resource.resourceType` and then cast `entry.resource` to the appropriate type.

For example:

```
if (entry.resource.resourceType === 'StructureDefinition') {
  const structureDefinition: StructureDefinition = <StructureDefinition> entry.resource;
  ...
}
```

Reply:
Yes, I did cast to StructureDefinition. But, the StructureDefinition does not have these properties defined/inheriting. This is how I casted. 
const structureDefinition = <StructureDefinition> entry.resource;

Reply:
there is no strong typing in `const structureDefinition = entry.resource;` 
`structureDefinition` in this case is not type-specific... it's just the variable name. So, it is inheriting whatever the type is that it is set from, which in this case is `entry.resource`.
`entry.resource` is of type `DomainResource`... _not_ of type `StructureDefinition`.

What you need instead is: `const structureDefinition: StructureDefinition = <StructureDefinition> entry.resource;`

Reply:
@seanmcilvenna, I tried this way today morning and received the same error.

Reply:
Let me try again.

Reply:
Same error @seanmcilvenna

Reply:
can you send me an updated code snippet?

Reply:
or commit your changes in their current state? so I can see them reflected in this PR?

Reply:
@seanmcilvenna I committed the changes with current state. Please check